### PR TITLE
Facade support.

### PR DESCRIPTION
**Pull request by bweston92**
# Definition of Done

- [x] We have all 3 facades working just by adding the "service provider".
- [x] We have appropriate tests in place.
- [x] There is an example for each use case in a new directory `/examples`.
- [x] A stub class that will make IDEs auto complete for all facades.

### CODE REVIEW

**Comment by bweston92** [comment]
@codeliner @prolic @sandrokeil - can you check this over.

File: .php_cs [hunk 1]
```diff
@@ -1,37 +1,6 @@
...
-        'unused_use',
-        'visibility',
-        'whitespacy_lines',
-    )
-);
-$config->finder($finder);
-return $config;
+$config = new Prooph\CS\Config\Prooph();
+$config->getFinder()->in(__DIR__ . '/src')->in(__DIR__ . '/tests');
+$cacheDir = getenv('TRAVIS') ? getenv('HOME') . '/.php-cs-fixer' : __DIR__;
```
Comment (bweston92) [inline .php_cs:4]: @prolic I have taken this from event-sourcing, how ever this directory "/home/travis/.php-cs-fixer" causes issues, any ideas?
https://travis-ci.org/prooph/laravel-package/jobs/230918730#L472

Comment (prolic) [reply .php_cs:4]: see here:
https://github.com/prooph/event-store/blob/master/.travis.yml#L21-L22

File: ide_stubs.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * prooph (http://getprooph.org/)
+ *
+ * @see       https://github.com/prooph/laravel-package for the canonical source repository
+ * @copyright Copyright (c) 2016 prooph software GmbH (http://prooph-software.com/)
+ * @license   https://github.com/prooph/laravel-package/blob/master/LICENSE.md New BSD License
+ */
+
```
Comment (sandrokeil) [inline ide_stubs.php]: @bweston92 Why do we need this?

**Comment by sandrokeil** [comment]
For issue #15

File: .php_cs (hunk 1 above)
Comment (bweston92) [reply .php_cs:4]: Thanks and updated.

**Comment by bweston92** [comment]
This PR if it passes now should be ready to merge. Still not install-able until #18 is finished merged.

**Pull request merged**